scheduler.py

```diff
@@ -1,173 +0,0 @@
-TICKER_BLACKLIST = ['A', 'ALL', 'ARE', 'AT', 'CEO', 'DD', 'EDIT', 'EV', 'FOR', 'ON', 'ONE', 'OR', 'RH', 'VERY']
-
-# Information we look for in Yahoo Finance
-YF_INFO_FIELDS = ['industry', 'previousClose', 'fiftyDayAverage', 'averageDailyVolume10Day']
-
-
-# Reads, parses and converts the ticker files into dictionary
-def build_dictionary(file_name, ticker_dict):
-    with open(file_name, 'r') as file:
-        for line in file:
-            line_list = line.strip().split('\t')
-            if len(line_list) < 2:
-                continue
-            symbol = line_list[0]
-            name = line_list[1]
-            if symbol not in ticker_dict:
-                ticker_dict[symbol] = name
-
-
-# Counts the tickers appeared in the titles and/or contents of the hot posts
-# Returns a dictionary with top 20 counts in non-increasing order
-def count_tickers(reddit, ticker_dict):
-    ticker_count = {}
-    hot_posts = reddit.subreddit('wallstreetbets').hot(limit=500)
-
-    for post in hot_posts:
-        word_in_title = post.title.split()
-        add_to_dict(word_in_title, ticker_dict, ticker_count)
-        word_in_content = post.selftext.split()
-        add_to_dict(word_in_content, ticker_dict, ticker_count)
-
-    sorted_ticker_count = dict(
-        sorted(ticker_count.items(), key=lambda item: item[1], reverse=True)[:TOTAL_NUMBER_OF_STOCKS])
-
-    return sorted_ticker_count
-
-
-# Helper function to aggregate the count of each ticker in ticker_count
-# Filters out tickers in the blacklist
-def add_to_dict(word_in_post, ticker_dict, ticker_count):
-    for word in word_in_post:
-        clean_word = word.strip()
-        if clean_word in TICKER_BLACKLIST:
-            continue
-        if clean_word in ticker_dict:
-            ticker_count[clean_word] = ticker_count.get(clean_word, 0) + 1
-
-
-# Helper function to determine if Yahoo Finance can provide the relevant information
-# If not, we skip this stock
-def found_all_fields(stock_info):
-    for field in YF_INFO_FIELDS:
-        if field not in stock_info.keys():
-            return False
-
-    return True
-
-
-# Helper function to get the most recent Top 10 stock record from the database
-# Returns a list of ticker symbols in non-decreasing order
-def get_prev_stock_list(collection, now):
-    prev_entry = collection.find({'timeStamp': {'$lte': now}}).sort([('timeStamp', -1)]).limit(1)
-
-    prev_time_stamp = prev_entry[0].get('timeStamp')
-    prev_stock_obj_list = collection.find({'timeStamp': prev_time_stamp})
-
-    prev_stock_list = []
-    for stock_obj in prev_stock_obj_list:
-        prev_stock_list.append(stock_obj.get('ticker'))
-
-    return prev_stock_list
-
-
-# Helper function to calculate ranking changes for the current Top 10 stocks
-# returns a tuple of whether this stock is new on the list and the ranking change (if applicable)
-def get_ranking_change(prev_stock_list, output_ticker, to_DB):
-    is_new = False
```

Comment:
is_new = False
ranking_change = 0
if ticker not in prev_stock_list:
    is_new = True
    return is_new, ranking_change

ranking_change = len(to_DB) - prev_stock_list.index(output_ticker)
return is_new, ranking_change